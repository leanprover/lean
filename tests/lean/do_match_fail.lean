open tactic

meta def no_intros : tactic unit :=
do [] ← intros, return ()

example (p q : Prop) : p → q → p :=
begin
  no_intros -- Should fail
end

example (p q : Prop) : p → q → p :=
begin
  intros,
  no_intros,  -- Should work
  assumption
end

example : list nat :=
do l    ← return $ [1, 2, 3],
   h::l ← return l,  -- non-exhaustive set of equations
   return 1

meta example : tactic nat :=
do (a, b) ← return (1, 2), -- Ok
   return a
