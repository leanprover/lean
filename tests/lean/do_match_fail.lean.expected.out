do_match_fail.lean:8:2: error: match failed
state:
p q : Prop,
a : p,
a_1 : q
⊢ p
do_match_fail.lean:9:0: error: failed
state:
p q : Prop,
a : p,
a_1 : q
⊢ p
do_match_fail.lean:20:3: error: invalid non-exhaustive set of equations (use 'set_option trace.eqn_compiler.elim_match true' for additional details)
